electronics_lib/DcDcConverters.py

```diff
@@ -1,5 +1,85 @@
 from electronics_abstract_parts import *
 
+class Tps61023_Device(DiscreteChip, FootprintBlock):
+  @init_in_parent
+  def __init__(self, current_draw: RangeLike = RangeExpr()):
+    super().__init__()
+    self.vin = self.Port(VoltageSink(
+      voltage_limits=(0.5, 5.5)*Volt,
+      current_draw=current_draw
+    ))
+    self.gnd = self.Port(Ground())
+    self.sw = self.Port(VoltageSource())
+    self.fb = self.Port(AnalogSink(impedance=(8000, float('inf')) * kOhm))
+    self.vout = self.Port(VoltageSource())
+
+  def contents(self) -> None:
+    super().contents()
+    self.footprint(
+      'U', 'Package_TO_SOT_SMD:SOT-23-6',
+      {
+        '1': self.fb,
+        '2': self.vin, # en
+        '3': self.vin,
+        '4': self.gnd,
+        '5': self.sw,
+        '6': self.vout,
+      },
+      mfr='Texas Instruments', part='TPS61023',
+      datasheet='https://www.ti.com/lit/ds/symlink/tps61023.pdf'
+    )
+
+class Tps61023(FootprintBlock):
```

Comment:
I recommend the `Ap3012` class as a modeling / starting point, and using the `DiscreteBoostConverter` superclass (but note one difference: the AP3012 is non-synchronous - it has a external blocking diode instead of an active switch that is integrated into the chip).

Also, note that the `Ap3012` uses a generator to size components from standard power converter equations, instead of using the suggested (generic) design. I'd also recommend using the generator approach here instead of static parameters - generators can better tailor a design to specific requirements. Can discuss more about power converters if you're interested.